src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Tree/Dependencies/Models/DependencyId.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Tree.Dependencies.Models
+{
+    internal readonly struct DependencyId : IEquatable<DependencyId>
+    {
+        public string ProviderId { get; }
+        public string ModelId { get; }
+
+        public DependencyId(string providerId, string modelId)
+        {
+            ProviderId = providerId;
+            ModelId = modelId;
+        }
+
+        public bool Equals(DependencyId other)
+        {
+            return StringComparers.DependencyProviderTypes.Equals(ProviderId, other.ProviderId) &&
+                   StringComparers.DependencyTreeIds.Equals(ModelId, other.ModelId);
+        }
+
+        public override bool Equals(object? obj) => obj is DependencyId other && Equals(other);
+
+        public override int GetHashCode()
+        {
+            return unchecked(StringComparers.DependencyProviderTypes.GetHashCode(ProviderId) * 397) ^ StringComparers.DependencyTreeIds.GetHashCode(ModelId);
```

Comment:
>397 [](start = 95, length = 3)

Where does this particular constant come from?

Reply:
R# generated this code. It's a prime that's large enough to overflow the first value a bit and distribute bits with the next value.